src/universal-dapp.js

```diff
@@ -18,14 +17,13 @@ function UniversalDApp (contracts, options, transactionDebugger) {
 
   self.web3 = options.web3;
   self.transactionDebugger = transactionDebugger;
+
   if (options.mode === 'vm') {
     // FIXME: use `options.vm` or `self.vm` consistently
     options.vm = true;
 
     self.accounts = {};
-
-    self.vm = new EthJSVM(null, null, { activatePrecompiles: true, enableHomestead: true });
```

Comment:
The big problem with wrapping it into `web3` is that the VM supports a much more advanced interface, which due to RPC limitations, cannot exists in `web3`.

There's no way to get detailed execution and output information from web3.